### DISCUSSION TITLE
complex result

### DISCUSSION THREAD

**Discussion by a-Rahmati** [Qalculate/qalculate-gtk#297] [Q&A] [answered]
Hello.
Qalculate can solve equations but just returns real results. Is there any option that shows the complex results?

![Qalculate!_084](https://user-images.githubusercontent.com/52320192/122419945-8b141c00-cfa0-11eb-8605-17860aa26873.png)

I want Qalculate to show `i, -i`.

Thanks.

**Comment by hanna-kn** [comment, marked as answer]
Select **Mode➞Assumption➞Number** (by default _x_ is assumed real), or use the _solve()_ function (e.g. `solve(x^2+1=0)`) which modifies the assumptions if no solutions are found. It is also possible to modify the assumptions of only _x_ using the menu of the _x_ button on the keypad

**Reply by a-Rahmati** [reply]
Thanks
